ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -126,7 +126,10 @@ public Bytes compute(final Bytes input, final MessageFrame messageFrame) {
     final BytesValueRLPInput bytesValueRLPInput =
         new BytesValueRLPInput(
             Bytes.wrap(Base64.getDecoder().decode(receiveResponse.getPayload())), false);
-    final PrivateTransaction privateTransaction = PrivateTransaction.readFrom(bytesValueRLPInput);
+    bytesValueRLPInput.enterList();
```

Comment:
We cannot assume that the payload is a list at this point, as we might have private Txs in Orion, or we might have besus that are on the older version.
I think the best would be if all payloads ending up here were rlp encoded as before, without the list!

Reply:
yes you are right!

Reply:
i need like an enclave payload handler thing because there are 3 types of payloads now - the normal ones, versioned ones and blobs of data...